src/app/components/filebrowseruss/filebrowseruss.component.ts

```diff
@@ -562,6 +586,34 @@ export class FileBrowserUSSComponent implements OnInit, OnDestroy {//IFileBrowse
     });
   }
 
+  toggleSearch() {
+    this.showSearch = !this.showSearch;
+    if (this.showSearch) {
+      this.focusSearchInput();
+      this.dataCached = _.cloneDeep(this.data); // We want a deep clone so we can modify this.data w/o changing this.dataCached
```

Comment:
In case if the search input is not empty, it would be good to call `this.searchInputChanged(this.searchInputCtrl.value)`